demosplan/DemosPlanCoreBundle/Logic/OwnsProcedureConditionFactory.php

```diff
@@ -13,120 +13,154 @@
-            $planningAgencyIsAuthorized = $this->isAuthorizedViaPlanningAgency();
-            $userIsAuthorized = $this->conditionFactory->propertyHasAnyOfValues(
-                $this->procedure->getAuthorizedUserIds(),
-                ['id']
-            );
-
-            $orgaOwnsProcedure = $this->conditionFactory->anyConditionApplies(
-                $userIsAuthorized,
-                $planningAgencyIsAuthorized
-            );
-        }
-
-        return $orgaOwnsProcedure;
+        // T8427: allow access by manually configured users if the config is set to `true`,
+        // overwriting the organisation-based access
+        return $this->globalConfig->hasProcedureUserRestrictedAccess()
+            ? $this->userIsExplicitlyAuthorized()
+            : $this->userOwnsProcedureViaOrgaOfUserThatCreatedTheProcedure();
     }
 
     /**
-     * Returns a condition to match users having the roles to theoretically own a procedure.
+     * Returns a condition to match users having the roles in the given customer to theoretically own a procedure.
      *
-     * @return FunctionInterface<bool>
+     * @return list<FunctionInterface<bool>>
      */
-    public function hasProcedureAccessingRole(): FunctionInterface
+    public function hasProcedureAccessingRole(Customer $customer): array
     {
-        $ownsOrgaRoleCondition = $this->conditionFactory->false();
+        $relevantRoles = [
+            RoleInterface::CUSTOMER_MASTER_USER,
+            ...User::PLANNING_AGENCY_ROLES,
+            ...User::HEARING_AUTHORITY_ROLES,
+        ];
 
-        if (null !== $this->procedure->getOrgaId()) {
+        if ($this->userOrProcedure instanceof User) {
+            $user = $this->userOrProcedure;
+
+            return $user->hasAnyOfRoles($relevantRoles, $customer)
+                ? [$this->conditionFactory->true()]
+                : [$this->conditionFactory->false()];
+        }
+
+        $procedure = $this->userOrProcedure;
+
+        if (null !== $procedure->getOrgaId()) {
             $this->logger->debug('Permissions: Check whether orga owns procedure');
-            // Fachplaner-Admin GLAUTH Kommune oder Fachplaner SB
-
-            // Fachplaner admin oder Fachplaner Sachbearbeiter oder Plattform-Admin oder AHB-Admin
-
-            $ownsOrgaRoleCondition = $this->conditionFactory->propertyHasAnyOfValues(
-                [
-                    Role::CUSTOMER_MASTER_USER,
-                    Role::PLANNING_AGENCY_ADMIN,
-                    Role::PLANNING_AGENCY_WORKER,
-                    Role::HEARING_AUTHORITY_ADMIN,
-                    Role::HEARING_AUTHORITY_WORKER,
-                ],
-                ['roleInCustomers', 'role', 'code']
-            );
+            $ownsOrgaRoleCondition = [
+                $this->conditionFactory->propertyHasAnyOfValues($relevantRoles, ['roleInCustomers', 'role', 'code']),
+                $this->isUserInCustomer($customer),
+            ];
+        } else {
```

Comment:
You can avoid this else by setting this as default value of the variable

Reply:
I'm fine with that for simple cases like `null` or primitive types as default, but here a method is called and an object would be created, which is instantly thrown away if the `if` branch is entered. This does not seem like a good design choice, as a programmer not familiar with this code may feel inclined to check for side effects in the logic of `$this->conditionFactory->false()` even though he is interested in the `if` case only.